integrations/sonarqube/main.py

```diff
@@ -18,14 +25,73 @@ def init_sonar_client() -> SonarQubeClient:
     )
 
 
+def produce_project_params(
+    api_filter: SonarQubeProjectApiFilter | None,
+) -> dict[str, Any]:
+    project_params: dict[str, Any] = {}
+    if not api_filter:
+        return project_params
+
+    if api_filter.analyzed_before:
+        project_params["analyzedBefore"] = api_filter.analyzed_before
+
+    if api_filter.on_provisioned_only:
+        project_params["onProvisionedOnly"] = api_filter.on_provisioned_only
+
+    if api_filter.projects:
+        project_params["projects"] = ",".join(api_filter.projects)
+
+    if api_filter.qualifiers:
+        project_params["qualifiers"] = ",".join(api_filter.qualifiers)
+
+    return project_params
+
+
+def produce_component_params(
+    client: SonarQubeClient, selector: Any, initial_params: dict[str, Any] = {}
+) -> dict[str, Any]:
+    component_query_params: dict[str, Any] = {}
+    if client.organization_id:
+        component_query_params["organization"] = client.organization_id
+
+    ## Handle query_params based on environment
+    if client.is_onpremise:
+        if initial_params:
+            component_query_params.update(initial_params)
+        elif event.resource_config:
+            # This might be called from places where event.resource_config is not set
+            # like on_start() when creating webhooks
+
+            selector = cast(CustomSelector, event.resource_config.selector)
+            component_query_params.update(selector.generate_request_params())
+
+    # remove project query params
+    for key in ["analyzedBefore", "onProvisionedOnly", "projects", "qualifiers"]:
+        component_query_params.pop(key, None)
+    return component_query_params
+
+
 sonar_client = init_sonar_client()
 
 
 @ocean.on_resync(ObjectKind.PROJECTS)
 async def on_project_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
     logger.info(f"Listing Sonarqube resource: {kind}")
+
     fetched_projects = False
-    async for project_list in sonar_client.get_all_projects():
+    selector = cast(SonarQubeProjectResourceConfig, event.resource_config).selector
+    sonar_client.metrics = selector.metrics
+
+    project_query_params = produce_project_params(selector.api_filters)
+
+    component_params = produce_component_params(sonar_client, selector)
+
```

Comment:
the get_all_project() should get one request param. You can pass either of these instead of passing both